addons/website/models/website.py

```diff
@@ -1474,7 +1481,17 @@ def _search_get_details(self, search_type, order, options):
 
         :return: list of search details obtained from the `website.searchable.mixin`'s `_search_get_detail()`
         """
-        result = []
+        if SEARCH_TYPE_MODELS:
+            # TODO: Remove in master.
+            if search_type == 'all':
+                model_names = reduce(operator.ior, SEARCH_TYPE_MODELS.values())
+            else:
+                model_names = SEARCH_TYPE_MODELS[search_type]
+            result = list(self.env[model_name]._search_get_detail(self, order, options)
+                    for model_name in model_names)
+        else:
+            result = []
```

Comment:
Ok to keep the `SEARCH_TYPE_MODELS` declaration but I would not keep this part, it will still crash in multi-tenancy with custo without the user especially knowing it.
At least this should be inside a try/except if you really want to keep it so if the model doesn't exists it doesn't crash.
But IMO removing it simply is fine too, I'll let you chose which one you prefer

Reply:
Added an `if model_name in self.env` instead of a `try/except`.